src/main/java/core/basesyntax/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public SupplyOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        if (fruitDao.containsFruit(fruitTransaction.getFruit())) {
+            fruitDao.update(fruitTransaction.getFruit(),
+                    fruitDao.get(fruitTransaction.getFruit()) + fruitTransaction.getQuantity());

```

Comment:
better create a DAO method for this, inside of which use map.merge with Integer::sum as function